src/Compilers/VisualBasic/Portable/Binding/Binder_ObjectInitializer.vb

```diff
@@ -508,6 +516,62 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return resultExpression
         End Function
 
+        Private Shared Sub CheckRequiredMembersInObjectInitializer(
+            constructor As MethodSymbol,
+            initializers As ImmutableArray(Of BoundExpression),
+            creationSyntax As SyntaxNode,
+            diagnostics As BindingDiagnosticBag)
+
+            If constructor Is Nothing OrElse constructor.HasSetsRequiredMembers Then
+                Return
+            End If
+
+            If constructor.ContainingType.HasRequiredMembersError Then
+                ' A use-site diagnostic will be reported on the use, so we don't need to do any more checking here.
+                Return
+            End If
+
+            Dim requiredMembers = constructor.ContainingType.AllRequiredMembers
+
+            If requiredMembers.Count = 0 Then
+                Return
+            End If
+
+            Dim requiredMembersBuilder = requiredMembers.ToBuilder()
+
+            If Not initializers.IsDefaultOrEmpty Then
+                For Each initializer In initializers
+                    Dim assignmentOperator = TryCast(initializer, BoundAssignmentOperator)
+                    If assignmentOperator Is Nothing Then
+                        Continue For
+                    End If
+
+                    Dim memberSymbol As Symbol = If(
+                        DirectCast(TryCast(assignmentOperator.Left, BoundPropertyAccess)?.PropertySymbol, Symbol),
+                        TryCast(assignmentOperator.Left, BoundFieldAccess)?.FieldSymbol)
+
+                    If memberSymbol Is Nothing Then
```

Comment:
>If memberSymbol Is Nothing Then [](http://example.com/codeflow?start=20&length=31)

Can we assert the receiver, since we are not checking it?

Reply:
I'm not certain what about the receiver you'd like me to assert?

Reply:
> I'm not certain what about the receiver you'd like me to assert?

Sorry, I meant an assert that the receiver is what we expect (placeholder?) of expected type.

Reply:
I guess I'm not sure what invariant you're hoping that this will verify.

Reply:
Nested member initializers on required members are an error in C# as well.

It appears that the receiver is not always present. Attributes do not pass one along, `ReceiverOpt` is nothing for that scenario.